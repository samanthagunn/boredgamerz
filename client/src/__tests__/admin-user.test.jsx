import React, { useState as useStateMock } from "react";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Users from "../components/admin-users"
import renderer from "react-test-renderer";
jest.mock("react", () => ({
    ...jest.requireActual("react"),
    useState: jest.fn(),
}));

window.confirm = jest.fn(() => true)
const setState = jest.fn();


    beforeAll(()=>{
        useStateMock.mockImplementation((init) => [init, setState]);
    })

    test("Users will render...",()=>{
        const { baseElement } = render(<Users/>);
        expect(baseElement).toBeDefined();
    })

    test("will match snapshot.", () => {
        const tree = renderer.create(<Users />).toJSON();
        expect(tree).toMatchSnapshot();
    })
