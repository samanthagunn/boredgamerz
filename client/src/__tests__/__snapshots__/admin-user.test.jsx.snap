// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will match snapshot. 1`] = `null`;
